libs/stats/odc/stats/_gm_ls_bitmask.py

```diff
@@ -0,0 +1,172 @@
+            |   |   |   |   |   |   |   |   |   |   o-------------------------> snow
+            |   |   |   |   |   |   |   |   |   x-----------------------------> clear
+            |   |   |   |   |   |   |   |   o---------------------------------> water
+            |   |   |   |   |   |   |   x-------------------------------------> cloud_confidence
+            |   |   |   |   |   |   o-----------------------------------------> cloud_confidence
+            |   |   |   |   |   x---------------------------------------------> cloud_shadow_confidence
+            |   |   |   |   o-------------------------------------------------> cloud_shadow_confidence
+            |   |   |   x-----------------------------------------------------> snow_ice_confidence
+            |   |   o---------------------------------------------------------> snow_ice_confidence
+            |   x-------------------------------------------------------------> cirrus_confidence
+            0-----------------------------------------------------------------> cirrus_confidence
+        """
+        mask_band = xx[self.mask_band]
+        xx = xx.drop_vars([self.mask_band])
+
+        # set cloud_mask (dilated_cloud + cloud + cloud_shadow) bitmask - True=cloud, False=non-cloud
+        cloud_mask = da.bitwise_and(mask_band, 0b0000_0000_0001_1010) != 0
+
+        # set no_data bitmask - True=data, False=no-data
+        keeps = da.bitwise_and(mask_band, 0b0000_0000_0000_0001) == 0
+
+        # drops nodata pixels and add cloud_mask from xx
+        xx = keep_good_only(xx, keeps)
+        xx["cloud_mask"] = cloud_mask
+
+        return xx
+
+    def input_data(self, task: Task) -> xr.Dataset:
+
+        chunks = {"y": -1, "x": -1}
+
+        xx = load_with_native_transform(
+            task.datasets,
+            bands=self.bands + [self.mask_band],
+            geobox=task.geobox,
+            native_transform=self._native_tr,
+            fuser=self._fuser,
+            groupby="solar_day",
+            resampling=self.resampling,
+            chunks=chunks,
+        )
+
+        return xx
+
+    def reduce(self, xx: xr.Dataset) -> xr.Dataset:
+        scale = 0.0000275
+        offset = -0.2
+        return_SR = False
+        cfg = dict(
+            maxiters=1000,
+            num_threads=1,
+            scale=scale,
+            offset=offset,
+            return_SR=return_SR,
+            reshape_strategy="mem",
+            out_chunks=(-1, -1, -1),
+            work_chunks=self.work_chunks,
+            compute_count=True,
+            compute_mads=True,
+        )
+
+        cloud_mask = xx["cloud_mask"]
+        xx = xx.drop_vars(["cloud_mask"])
+
+        # erase pixels with cloud
+        xx = erase_bad(xx, cloud_mask)
+
+        gm = geomedian_with_mads(xx, **cfg)
+        gm = gm.rename(self.renames)
+
+        # Scale USGS Landsat bands into surface reflectance
+        if return_SR:
+            for band in gm.data_vars:
```

Comment:
I think this could be:

```suggestion
            for band in self.bands:
```